yeonjae/programmers-2/파일명정렬.js

```diff
@@ -0,0 +1,29 @@
+function solution(files) {
+  files.sort((a, b) => {
+    const [aTotal, aHead, aNum] = a.match(/(^[^0-9]+)(\d+)(.*)/);
+    const [bTotal, bHead, bNum] = b.match(/(^[^0-9]+)(\d+)(.*)/);
+
+    if (aHead.toLowerCase() > bHead.toLowerCase()) {
+      return 1;
+    } else if (aHead.toLowerCase() < bHead.toLowerCase()) {
+      return -1;
+    }
+
+    if (+aNum > +bNum) {
+      return 1;
+    } else if (+aNum < +bNum) {
+      return -1;
+    }
+    return 0;
+  });
+
+  return files;
+}
```

Comment:
```suggestion
function solution(files) {
  const REGEX = /(^[^0-9]+)(\d+)(.*)/;
  
  return files.sort((a, b) => {
    const [aTotal, aHead, aNum] = a.match(REGEX);
    const [bTotal, bHead, bNum] = b.match(REGEX);
    const aLowerHead = aHead.toLowerCase();
    const bLowerHead = bHead.toLowerCase();
    
    return aLowerHead.localeCompare(bLowerHead) || +aNum - +bNum
  });
}
```
- 풀이잘봤습니다~
- 정규표현식은 상수로 빼서 활용해주세요
- sort내부에서 localeCompare 메서드와 || 연산자를 활용해서 깔끔한 코드가 가능합니다.

Reply:
localeCompare이건 처음알았네요! 코멘트 감사합니다